packages/flutter_tools/lib/executable.dart

```diff
@@ -68,28 +107,14 @@ Future<Null> main(List<String> args) async {
     args.removeWhere((String option) => option == '-v' || option == '--verbose');
   }
 
-  FlutterCommandRunner runner = new FlutterCommandRunner(verboseHelp: verboseHelp)
-    ..addCommand(new AnalyzeCommand(verboseHelp: verboseHelp))
-    ..addCommand(new BuildCommand(verboseHelp: verboseHelp))
-    ..addCommand(new ChannelCommand())
-    ..addCommand(new ConfigCommand())
-    ..addCommand(new CreateCommand())
-    ..addCommand(new DaemonCommand(hidden: !verboseHelp))
-    ..addCommand(new DevicesCommand())
-    ..addCommand(new DoctorCommand())
-    ..addCommand(new DriveCommand())
-    ..addCommand(new FormatCommand())
-    ..addCommand(new InstallCommand())
-    ..addCommand(new LogsCommand())
-    ..addCommand(new PackagesCommand())
-    ..addCommand(new PrecacheCommand())
-    ..addCommand(new RunCommand(verboseHelp: verboseHelp))
-    ..addCommand(new ScreenshotCommand())
-    ..addCommand(new StopCommand())
-    ..addCommand(new TestCommand())
-    ..addCommand(new TraceCommand())
-    ..addCommand(new UpdatePackagesCommand(hidden: !verboseHelp))
-    ..addCommand(new UpgradeCommand());
+  TopLevelOptions topLevelOptions = new TopLevelOptions(
+    verboseHelp: verboseHelp,
+  );
+
+  FlutterCommandRunner runner = new FlutterCommandRunner(verboseHelp: verboseHelp);
```

Comment:
Suggestion: define the following:

```
run({
  List<String> args,
  List<FlutterCommand> commands,
})
```

.. then have `main()` invoke `run()` with the default commands.  It saves you a mutable global, and it will make it slightly cleaner to pass a list of custom commands in Google's internal build system.